packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -458,6 +459,28 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
 
         // Update the used routes in this data store's summarizer node.
         this.summarizerNode.updateUsedRoutes(usedRoutes);
+
+        // If we are loaded, call the channel so it can update the used routes of the child contexts.
+        // If we are not loaded, we will update this when we are realized.
+        if (this.loaded) {
+            this.updateChannelUsedRoutes();
+        }
+    }
+
+    /**
+     * Calls the channel to update used routes of its child contexts.
+     */
+    private updateChannelUsedRoutes() {
+        assert(this.channel !== undefined, "Channel should not be undefined when updating used routes");
+        // Remove the route to this data store, if it exists.
+        const usedChannelRoutes = this.summarizerNode.usedRoutes.filter(
+            (id: string) => { return id !== "/" && id !== ""; },
+        );
```

Comment:
All of this code could be enclosed in the `if (this.channel.updateUsedRoutes !== undefined)` condition.